src/lib.rs

```
@@ -1,37 +1,116 @@
+            let mut arr = [0u8; PUBLIC_KEY_LENGTH];
+            for (place, element) in arr.iter_mut().zip(v.iter()) {
+                *place = *element;
+            }
+            Ok(Sha512PublicKey(arr))
+        } else {
+            Err(format!(
+                "Sha512PublicKey: Wrong vector length. Expected {}. Got {}",
+                PUBLIC_KEY_LENGTH,
+                v.len()
+            ))
+        }
+    }
+}
+
+impl TryFrom<Vec<u8>> for Sha512PrivateKey {
+    type Error = String;
+
+    fn try_from(v: Vec<u8>) -> Result<Self, Self::Error> {
+        if v.len() == PRIVATE_KEY_LENGTH {
+            let mut arr = [0u8; PRIVATE_KEY_LENGTH];
+            for (place, element) in arr.iter_mut().zip(v.iter()) {
+                *place = *element;
+            }
+            Ok(Sha512PrivateKey(arr))
+        } else {
+            Err(format!(
+                "Sha512PrivateKey: Wrong vector length. Expected {}. Got {}",
+                PRIVATE_KEY_LENGTH,
+                v.len()
+            ))
+        }
+    }
+}
+
 /// This library contains utilities to work with cryptographical keys.
 
 /// If you need a keypair, use this method by default.
 /// It uses [Ursa](https://docs.rs/ursa) library with `Ed25519` solution of [EdDSA](https://tools.ietf.org/html/rfc8032).
 /// [OsRng](https://docs.rs/rand/0.7/rand/rngs/struct.OsRng.html) random number generator is used.
-pub fn generate_keypair() -> Result<(Vec<u8>, Vec<u8>), String> {
+pub fn generate_keypair() -> Result<(Sha512PublicKey, Sha512PrivateKey), String> {
     match Ed25519Sha512.keypair(Option::None) {
-        Ok((public_key, private_key)) => Ok((public_key.0.clone(), private_key.0.clone())),
+        Ok((public_key, private_key)) => {
+            let public_key = Sha512PublicKey::try_from(public_key.0.clone()).unwrap();
+            let private_key = Sha512PrivateKey::try_from(private_key.0.clone()).unwrap();
+            Ok((public_key, private_key))
+        }
```

Let's clean it up.
1. Do we really need temporary variables with confusing names? If we need them, let's rename to sha_512_public/private_key.
2. Do not use unwrap and punish me if you see when I use it (maybe only tests should be an excuse). It will fail in runtime and without a good error message. At least `expect` should be used, but better to propagate via `?` or match result.
3. Maybe instead of `public_key.0.clone()` we should use something like `&public_key.0` now? What do you think?

---

The reason I used `unwrap` there is that we rely on the underlying library to have exact sizes, and it would be reasonable to panic otherwise. 

But maybe yes, we can propagate the error.

---

Maybe it's okay to panic, but please use `expect` in this case with a good panic message.